sourcegen-generator-java/src/main/java/io/micronaut/sourcegen/JavaPoetSourceGenerator.java

```diff
@@ -583,15 +589,39 @@ private CodeBlock renderExpression(@Nullable ObjectDef objectDef, MethodDef meth
                 andExpressionDef.left() instanceof ExpressionDef.And || andExpressionDef.left() instanceof ExpressionDef.Or;
             boolean isRightCondition = andExpressionDef.right() instanceof ExpressionDef.Condition ||
                 andExpressionDef.right() instanceof ExpressionDef.And || andExpressionDef.right() instanceof ExpressionDef.Or;
-            return CodeBlock.concat(
-                (isLeftCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
-                renderExpression(objectDef, methodDef, andExpressionDef.left()),
-                (isLeftCondition) ? CodeBlock.of(")") : CodeBlock.of(""),
-                CodeBlock.of(" && "),
-                (isRightCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
-                renderExpression(objectDef, methodDef, andExpressionDef.right()),
-                (isRightCondition) ? CodeBlock.of(")") : CodeBlock.of("")
-            );
+            if (isLeftCondition && isRightCondition) {
```

Comment:
This looks too complicated, can you just have List and add codeblock that are needed then call concat

Reply:
Like my previous implementation?
```
if (expressionDef instanceof ExpressionDef.And andExpressionDef) {
            boolean isLeftCondition = andExpressionDef.left() instanceof ExpressionDef.Condition ||
                andExpressionDef.left() instanceof ExpressionDef.And || andExpressionDef.left() instanceof ExpressionDef.Or;
            boolean isRightCondition = andExpressionDef.right() instanceof ExpressionDef.Condition ||
                andExpressionDef.right() instanceof ExpressionDef.And || andExpressionDef.right() instanceof ExpressionDef.Or;
            return CodeBlock.concat(
                (isLeftCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
                renderExpression(objectDef, methodDef, andExpressionDef.left()),
                (isLeftCondition) ? CodeBlock.of(")") : CodeBlock.of(""),
                CodeBlock.of(" && "),
                (isRightCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
                renderExpression(objectDef, methodDef, andExpressionDef.right()),
                (isRightCondition) ? CodeBlock.of(")") : CodeBlock.of("")
            );
 }
```
I changed this according to this review: https://github.com/micronaut-projects/micronaut-sourcegen/pull/149#discussion_r1778742157

Reply:
Something like this:
```
        List<CodeBlock> codeBlocks = new ArrayList<>();
            if (isLeftCondition) {
                codeBlocks.add(CodeBlock.of("("));
                codeBlocks.add(renderExpression(objectDef, methodDef, andExpressionDef.left()));
                codeBlocks.add(CodeBlock.of(")"));
            } else {
                codeBlocks.add(renderExpression(objectDef, methodDef, andExpressionDef.left());
            }
```
you can also create a new method to reduce the duplicity of checking for left and right